scripts/bootloader/provision.py

```diff
@@ -13,13 +13,15 @@
 from hashlib import sha256
 
 
-def generate_provision_hex_file(s0_address, s1_address, hashes, provision_address, output):
+def generate_provision_hex_file(s0_address, s1_address, hashes, provision_address, output, num_counter_slots):
     # Add addresses
     provision_data = struct.pack('III', s0_address, s1_address, len(hashes))
     for mhash in hashes:
         provision_data += struct.pack('I', 0xFFFFFFFF) # Invalidation token
         provision_data += mhash
 
+    provision_data += struct.pack('I', int(num_counter_slots, 0))
```

Comment:
Is each slot here a half-word or word? `I` is 4 byte.

Reply:
Why is this not set to 0xFFFFF?

Reply:
This is the number of slots (`struct monotonic_counter::num_counter_slots`), it's not the slots themselves.

Reply:
Ah, should have spotted that.